lib/resty/redis.lua

```diff
@@ -256,15 +256,10 @@ local function _gen_req(args)
         local arg = args[i]
         nbits = nbits + 1
 
-        if not arg then
-            req[nbits] = "$-1\r\n"
-
-        else
-            if type(arg) ~= "string" then
-                arg = tostring(arg)
-            end
-            req[nbits] = "$" .. #arg .. "\r\n" .. arg .. "\r\n"
+        if type(arg) ~= "string" then
+            arg = tostring(arg)
         end
+        req[nbits] = "$" .. #arg .. "\r\n" .. arg .. "\r\n"
```

Comment:
Let's avoid string concatenation here. Just insert them into different slots in the `req` table :)